homeassistant/loader.py

```diff
@@ -54,10 +54,38 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+#
+# Integration.get_component will check preload platforms and
+# try to import the code to avoid a thundering heard of import
+# executor jobs later in the startup process.
+#
+# default platforms are prepopulated in this list to ensure that
+# by the time the component is loaded, we check if the platform is
+# available.
+#
+# This list can be extended by calling async_register_preload_platform
+#
+BASE_PRELOAD_PLATFORMS = [
+    "config",
+    "diagnostics",
+    "energy",
```

Comment:
this is used by 1 integration, worth checking for each integration? Or is it worth, because the way the solar forecast is set up using `async_process_integration_platforms`? I wonder if we should make an on-demand version of that helper instead, given that solar forecasts are never used at startup Same for diagnostics, logbook etc.

Reply:
Once `async_process_integration_platforms` gets called with any platform every integration has to check them eventually so its better to do it all in  single executor job.

Reply:
I limited the platforms down and than reverted that change because it doesn't actually try to load the platform if it doesn't exist and it cheaper to cache that it doesn't exist vs trying to have to load it later

Reply:
Even if we take them out of the list they will always have to pay for the `stat`s later as integration platforms are unconditionally checked per integration

If the list gets too long, there is a point where `os.listdir` makes more sense than the `stat()s`

Reply:
Rewrote this to use listdir

Reply:
It also means we can call `platforms_exists` in the loop or in the executor as its now thread safe and it means https://github.com/home-assistant/core/pull/112168 was able to delete about 40% of the new code